gapir/cc/interpreter.cpp

```diff
@@ -469,6 +469,22 @@ Interpreter::Result Interpreter::switchThread(uint32_t opcode) {
   return CHANGE_THREAD;
 }
 
+Interpreter::Result Interpreter::addJumpLabel(uint32_t opcode) {
+  auto jump_id = extract26bitData(opcode);
+  mJumpLables[jump_id] = mCurrentInstruction;
+  return SUCCESS;
+}
+
+Interpreter::Result Interpreter::jumpNZ(uint32_t opcode) {
+  auto jump_id = extract26bitData(opcode);
+  auto should_jump = mStack.pop<int32_t>();
+  if (should_jump != 0) {
+    mCurrentInstruction = mJumpLables[jump_id];
+  }
+
```

Comment:
Should we assert the stack is zero length here? Carrying across stack values seems like a really bad idea.